{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"FeatureExtraction.ipynb","provenance":[],"collapsed_sections":[],"authorship_tag":"ABX9TyPmDfr06WoKgM4wCRQHyDqi"},"kernelspec":{"name":"python3","display_name":"Python 3"},"accelerator":"GPU"},"cells":[{"cell_type":"code","metadata":{"id":"chm-eWMO_LKL","colab_type":"code","outputId":"5d07d6d9-605f-4fc7-841f-7d23905e0439","executionInfo":{"status":"ok","timestamp":1588366047275,"user_tz":-330,"elapsed":75847,"user":{"displayName":"Mahantesh ambi","photoUrl":"","userId":"14442223101884743872"}},"colab":{"base_uri":"https://localhost:8080/","height":1000}},"source":["!pip install git+https://github.com/rcmalli/keras-vggface.git\n","!pip install mtcnn\n","!pip install --upgrade cython\n","!pip install --upgrade setuptools\n","!pip install nmslib\n","!pip install pandas\n","!pip install cmake\n","!pip install dlib\n","!pip install imutils\n","!pip install face_recognition"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Collecting git+https://github.com/rcmalli/keras-vggface.git\n","  Cloning https://github.com/rcmalli/keras-vggface.git to /tmp/pip-req-build-9ep4d4bx\n","  Running command git clone -q https://github.com/rcmalli/keras-vggface.git /tmp/pip-req-build-9ep4d4bx\n","Requirement already satisfied: numpy>=1.9.1 in /usr/local/lib/python3.6/dist-packages (from keras-vggface==0.6) (1.18.3)\n","Requirement already satisfied: scipy>=0.14 in /usr/local/lib/python3.6/dist-packages (from keras-vggface==0.6) (1.4.1)\n","Requirement already satisfied: h5py in /usr/local/lib/python3.6/dist-packages (from keras-vggface==0.6) (2.10.0)\n","Requirement already satisfied: pillow in /usr/local/lib/python3.6/dist-packages (from keras-vggface==0.6) (7.0.0)\n","Requirement already satisfied: keras in /usr/local/lib/python3.6/dist-packages (from keras-vggface==0.6) (2.3.1)\n","Requirement already satisfied: six>=1.9.0 in /usr/local/lib/python3.6/dist-packages (from keras-vggface==0.6) (1.12.0)\n","Requirement already satisfied: pyyaml in /usr/local/lib/python3.6/dist-packages (from keras-vggface==0.6) (3.13)\n","Requirement already satisfied: keras-preprocessing>=1.0.5 in /usr/local/lib/python3.6/dist-packages (from keras->keras-vggface==0.6) (1.1.0)\n","Requirement already satisfied: keras-applications>=1.0.6 in /usr/local/lib/python3.6/dist-packages (from keras->keras-vggface==0.6) (1.0.8)\n","Building wheels for collected packages: keras-vggface\n","  Building wheel for keras-vggface (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Created wheel for keras-vggface: filename=keras_vggface-0.6-cp36-none-any.whl size=8311 sha256=66a98288fe58dfd3b3236650932fc464c9d85cd3201cac87be2391dbceda9300\n","  Stored in directory: /tmp/pip-ephem-wheel-cache-yt0227ct/wheels/36/07/46/06c25ce8e9cd396dabe151ea1d8a2bc28dafcb11321c1f3a6d\n","Successfully built keras-vggface\n","Installing collected packages: keras-vggface\n","Successfully installed keras-vggface-0.6\n","Collecting mtcnn\n","\u001b[?25l  Downloading https://files.pythonhosted.org/packages/67/43/abee91792797c609c1bf30f1112117f7a87a713ebaa6ec5201d5555a73ef/mtcnn-0.1.0-py3-none-any.whl (2.3MB)\n","\u001b[K     |████████████████████████████████| 2.3MB 9.6MB/s \n","\u001b[?25hRequirement already satisfied: opencv-python>=4.1.0 in /usr/local/lib/python3.6/dist-packages (from mtcnn) (4.1.2.30)\n","Requirement already satisfied: keras>=2.0.0 in /usr/local/lib/python3.6/dist-packages (from mtcnn) (2.3.1)\n","Requirement already satisfied: numpy>=1.11.3 in /usr/local/lib/python3.6/dist-packages (from opencv-python>=4.1.0->mtcnn) (1.18.3)\n","Requirement already satisfied: six>=1.9.0 in /usr/local/lib/python3.6/dist-packages (from keras>=2.0.0->mtcnn) (1.12.0)\n","Requirement already satisfied: keras-applications>=1.0.6 in /usr/local/lib/python3.6/dist-packages (from keras>=2.0.0->mtcnn) (1.0.8)\n","Requirement already satisfied: keras-preprocessing>=1.0.5 in /usr/local/lib/python3.6/dist-packages (from keras>=2.0.0->mtcnn) (1.1.0)\n","Requirement already satisfied: pyyaml in /usr/local/lib/python3.6/dist-packages (from keras>=2.0.0->mtcnn) (3.13)\n","Requirement already satisfied: scipy>=0.14 in /usr/local/lib/python3.6/dist-packages (from keras>=2.0.0->mtcnn) (1.4.1)\n","Requirement already satisfied: h5py in /usr/local/lib/python3.6/dist-packages (from keras>=2.0.0->mtcnn) (2.10.0)\n","Installing collected packages: mtcnn\n","Successfully installed mtcnn-0.1.0\n","Requirement already up-to-date: cython in /usr/local/lib/python3.6/dist-packages (0.29.17)\n","Requirement already up-to-date: setuptools in /usr/local/lib/python3.6/dist-packages (46.1.3)\n","Collecting nmslib\n","\u001b[?25l  Downloading https://files.pythonhosted.org/packages/d5/fd/7d7428d29f12be5d1cc6d586d425b795cc9c596ae669593fd4f388602010/nmslib-2.0.6-cp36-cp36m-manylinux2010_x86_64.whl (12.9MB)\n","\u001b[K     |████████████████████████████████| 13.0MB 1.5MB/s \n","\u001b[?25hCollecting pybind11>=2.2.3\n","\u001b[?25l  Downloading https://files.pythonhosted.org/packages/89/e3/d576f6f02bc75bacbc3d42494e8f1d063c95617d86648dba243c2cb3963e/pybind11-2.5.0-py2.py3-none-any.whl (296kB)\n","\u001b[K     |████████████████████████████████| 296kB 55.1MB/s \n","\u001b[?25hRequirement already satisfied: numpy>=1.10.0 in /usr/local/lib/python3.6/dist-packages (from nmslib) (1.18.3)\n","Requirement already satisfied: psutil in /usr/local/lib/python3.6/dist-packages (from nmslib) (5.4.8)\n","Installing collected packages: pybind11, nmslib\n","Successfully installed nmslib-2.0.6 pybind11-2.5.0\n","Requirement already satisfied: pandas in /usr/local/lib/python3.6/dist-packages (1.0.3)\n","Requirement already satisfied: python-dateutil>=2.6.1 in /usr/local/lib/python3.6/dist-packages (from pandas) (2.8.1)\n","Requirement already satisfied: numpy>=1.13.3 in /usr/local/lib/python3.6/dist-packages (from pandas) (1.18.3)\n","Requirement already satisfied: pytz>=2017.2 in /usr/local/lib/python3.6/dist-packages (from pandas) (2018.9)\n","Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.6/dist-packages (from python-dateutil>=2.6.1->pandas) (1.12.0)\n","Requirement already satisfied: cmake in /usr/local/lib/python3.6/dist-packages (3.12.0)\n","Requirement already satisfied: dlib in /usr/local/lib/python3.6/dist-packages (19.18.0)\n","Requirement already satisfied: imutils in /usr/local/lib/python3.6/dist-packages (0.5.3)\n","Collecting face_recognition\n","  Downloading https://files.pythonhosted.org/packages/1e/95/f6c9330f54ab07bfa032bf3715c12455a381083125d8880c43cbe76bb3d0/face_recognition-1.3.0-py2.py3-none-any.whl\n","Requirement already satisfied: Pillow in /usr/local/lib/python3.6/dist-packages (from face_recognition) (7.0.0)\n","Requirement already satisfied: numpy in /usr/local/lib/python3.6/dist-packages (from face_recognition) (1.18.3)\n","Collecting face-recognition-models>=0.3.0\n","\u001b[?25l  Downloading https://files.pythonhosted.org/packages/cf/3b/4fd8c534f6c0d1b80ce0973d01331525538045084c73c153ee6df20224cf/face_recognition_models-0.3.0.tar.gz (100.1MB)\n","\u001b[K     |████████████████████████████████| 100.2MB 74kB/s \n","\u001b[?25hRequirement already satisfied: Click>=6.0 in /usr/local/lib/python3.6/dist-packages (from face_recognition) (7.1.2)\n","Requirement already satisfied: dlib>=19.7 in /usr/local/lib/python3.6/dist-packages (from face_recognition) (19.18.0)\n","Building wheels for collected packages: face-recognition-models\n","  Building wheel for face-recognition-models (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Created wheel for face-recognition-models: filename=face_recognition_models-0.3.0-py2.py3-none-any.whl size=100566172 sha256=00b01e7769df707da32db92ca20d1f781a1d3f60219201d135fed3cdafbcf035\n","  Stored in directory: /root/.cache/pip/wheels/d2/99/18/59c6c8f01e39810415c0e63f5bede7d83dfb0ffc039865465f\n","Successfully built face-recognition-models\n","Installing collected packages: face-recognition-models, face-recognition\n","Successfully installed face-recognition-1.3.0 face-recognition-models-0.3.0\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"T84bUWOjAry4","colab_type":"code","outputId":"8233f054-9b46-488a-a450-e170418cdddd","executionInfo":{"status":"ok","timestamp":1588366353527,"user_tz":-330,"elapsed":2206,"user":{"displayName":"Mahantesh ambi","photoUrl":"","userId":"14442223101884743872"}},"colab":{"base_uri":"https://localhost:8080/","height":34}},"source":["from google.colab import drive\n","drive.mount('/content/gdrive', force_remount=True)"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Mounted at /content/gdrive\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"WDBANddmCjk4","colab_type":"code","outputId":"86f9b8e6-0e7d-4106-9fa4-a6fb13c454ae","executionInfo":{"status":"ok","timestamp":1588326033000,"user_tz":-330,"elapsed":7453,"user":{"displayName":"Mahantesh ambi","photoUrl":"","userId":"14442223101884743872"}},"colab":{"base_uri":"https://localhost:8080/","height":102}},"source":["!ls /content/gdrive/My\\ Drive/FaceProcessing/\n","!ls /content/gdrive/My\\ Drive/FaceProcessing/code/"],"execution_count":0,"outputs":[{"output_type":"stream","text":["code\t\t     processed_images\t      requirements.txt\n","preprocessed_images  processed_square_images  steps.txt\n","FaceFeatures.hnsw     FeatureExtraction.ipynb  Untitled1.ipynb\n","FaceProcessing.ipynb  FeatureExtractor.py      Untitled2.ipynb\n","FaceSearch.ipynb      prepare_data\t       Untitled.ipynb\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"5GIPTNzIBPiI","colab_type":"code","outputId":"99e31a2a-149b-4d0c-a3fd-7bb99ce0b234","executionInfo":{"status":"ok","timestamp":1588362557667,"user_tz":-330,"elapsed":600054,"user":{"displayName":"Mahantesh ambi","photoUrl":"","userId":"14442223101884743872"}},"colab":{"base_uri":"https://localhost:8080/","height":1000}},"source":["!python /content/gdrive/My\\ Drive/FaceProcessing/code/FeatureExtractor.py -model_type \"vgg16\" -debug False -images /content/gdrive/My\\ Drive/FaceProcessing/processed_square_images/"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Using TensorFlow backend.\n","2020-05-01 19:39:20.432524: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1\n","Your CPU supports instructions that this binary was not compiled to use: SSE3 SSE4.1 SSE4.2 AVX AVX2\n","For maximum performance, you can install NMSLIB from sources \n","pip install --no-binary :all: nmslib\n","2020-05-01 19:39:21.844064: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1\n","2020-05-01 19:39:21.892900: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-01 19:39:21.893515: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: \n","pciBusID: 0000:00:04.0 name: Tesla P4 computeCapability: 6.1\n","coreClock: 1.1135GHz coreCount: 20 deviceMemorySize: 7.43GiB deviceMemoryBandwidth: 178.99GiB/s\n","2020-05-01 19:39:21.893572: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1\n","2020-05-01 19:39:22.107794: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10\n","2020-05-01 19:39:22.241730: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10\n","2020-05-01 19:39:22.267404: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10\n","2020-05-01 19:39:22.521000: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10\n","2020-05-01 19:39:22.571597: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10\n","2020-05-01 19:39:23.045744: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7\n","2020-05-01 19:39:23.045943: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-01 19:39:23.046653: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-01 19:39:23.047214: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1703] Adding visible gpu devices: 0\n","2020-05-01 19:39:23.070738: I tensorflow/core/platform/profile_utils/cpu_utils.cc:102] CPU Frequency: 2200000000 Hz\n","2020-05-01 19:39:23.070929: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x1387640 initialized for platform Host (this does not guarantee that XLA will be used). Devices:\n","2020-05-01 19:39:23.070959: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version\n","2020-05-01 19:39:23.207202: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-01 19:39:23.207689: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x6324000 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:\n","2020-05-01 19:39:23.207724: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Tesla P4, Compute Capability 6.1\n","2020-05-01 19:39:23.208846: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-01 19:39:23.209230: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: \n","pciBusID: 0000:00:04.0 name: Tesla P4 computeCapability: 6.1\n","coreClock: 1.1135GHz coreCount: 20 deviceMemorySize: 7.43GiB deviceMemoryBandwidth: 178.99GiB/s\n","2020-05-01 19:39:23.209300: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1\n","2020-05-01 19:39:23.209338: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10\n","2020-05-01 19:39:23.209365: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10\n","2020-05-01 19:39:23.209389: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10\n","2020-05-01 19:39:23.209409: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10\n","2020-05-01 19:39:23.209430: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10\n","2020-05-01 19:39:23.209453: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7\n","2020-05-01 19:39:23.209528: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-01 19:39:23.209912: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-01 19:39:23.210241: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1703] Adding visible gpu devices: 0\n","2020-05-01 19:39:23.214328: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1\n","2020-05-01 19:39:29.668118: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1102] Device interconnect StreamExecutor with strength 1 edge matrix:\n","2020-05-01 19:39:29.668170: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1108]      0 \n","2020-05-01 19:39:29.668198: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1121] 0:   N \n","2020-05-01 19:39:29.672612: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-01 19:39:29.673052: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-01 19:39:29.673435: W tensorflow/core/common_runtime/gpu/gpu_bfc_allocator.cc:39] Overriding allow_growth setting because the TF_FORCE_GPU_ALLOW_GROWTH environment variable is set. Original config value was 0.\n","2020-05-01 19:39:29.673482: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1247] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 6966 MB memory) -> physical GPU (device: 0, name: Tesla P4, pci bus id: 0000:00:04.0, compute capability: 6.1)\n","2020-05-01 19:39:34.756539: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7\n","2020-05-01 19:39:39.168544: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10\n","(337, -27, 474, 124) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000014.jpg\n","(171, -13, 223, 50) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000035.jpg\n","(-9, 110, 66, 210) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000057.jpg\n","(79, -13, 401, 405) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000050.jpg\n","(383, 66, 513, 226) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000049.jpg\n","(-4, 113, 117, 280) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000049.jpg\n","(397, -29, 532, 151) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000074.jpg\n","(427, -8, 512, 111) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000072.jpg\n","(251, -27, 333, 68) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000072.jpg\n","(303, -20, 399, 88) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000075.jpg\n","Extracted 100 faces in [74.145] seconds \n","(390, 133, 520, 326) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000110.jpg\n","(369, 65, 522, 303) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000170.jpg\n","(-7, 83, 87, 228) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000177.jpg\n","(-21, 59, 151, 305) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000204.jpg\n","Extracted 200 faces in [65.881] seconds \n","(440, 31, 517, 130) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000215.jpg\n","(-7, 86, 122, 279) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000230.jpg\n","(-5, 53, 30, 112) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000250.jpg\n","(410, 58, 513, 198) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000242.jpg\n","(301, -13, 395, 102) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000247.jpg\n","(258, 371, 386, 530) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000245.jpg\n","(343, -1, 392, 60) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000264.jpg\n","(253, -6, 305, 60) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000260.jpg\n","(315, 62, 512, 340) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000300.jpg\n","Extracted 300 faces in [66.167] seconds \n","(-9, 112, 184, 370) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000320.jpg\n","(395, 435, 470, 525) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000380.jpg\n","Extracted 400 faces in [65.663] seconds \n","(-9, 89, 211, 408) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000419.jpg\n","(334, 90, 538, 398) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000419.jpg\n","(392, 462, 433, 517) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000430.jpg\n","(89, -7, 147, 60) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000423.jpg\n","(-1, -10, 36, 33) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000423.jpg\n","(-14, 91, 49, 178) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000423.jpg\n","(-12, 74, 140, 283) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000436.jpg\n","(-9, 232, 63, 359) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000467.jpg\n","(-1, 316, 32, 365) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000485.jpg\n","Extracted 500 faces in [65.173] seconds \n","(-5, 92, 229, 385) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000500.jpg\n","(291, 47, 519, 350) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000500.jpg\n","(-11, 106, 109, 289) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000493.jpg\n","(466, 170, 528, 267) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000512.jpg\n","(245, -5, 287, 40) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000530.jpg\n","(-5, 179, 96, 315) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000533.jpg\n","(108, -21, 189, 69) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000578.jpg\n","(205, -17, 506, 406) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000574.jpg\n","(457, 140, 513, 218) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000588.jpg\n","Extracted 600 faces in [64.768] seconds \n","(297, -14, 384, 96) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000600.jpg\n","(-5, 90, 94, 240) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000630.jpg\n","(-7, 131, 116, 311) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000659.jpg\n","(468, 82, 514, 149) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000678.jpg\n","MTCNN did not detect face for: /content/gdrive/My Drive/FaceProcessing/processed_square_images/000699.jpg\n","(219, 481, 258, 528) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000711.jpg\n","Extracted 700 faces in [64.980] seconds \n","(425, 91, 529, 249) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000755.jpg\n","(-2, 107, 79, 211) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000783.jpg\n","(475, 231, 524, 312) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000793.jpg\n","(173, -26, 440, 300) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000814.jpg\n","Extracted 800 faces in [65.130] seconds \n","MTCNN did not detect face for: /content/gdrive/My Drive/FaceProcessing/processed_square_images/000836.jpg\n","Extracted faces in 839 images at [556.263] seconds\n","Total 1530 faces detected\n","Preprocessed faces in [0.887] seconds: \n","Downloading data from https://github.com/rcmalli/keras-vggface/releases/download/v2.0/rcmalli_vggface_tf_notop_vgg16.h5\n","58916864/58909280 [==============================] - 6s 0us/step\n","Extracted features in [12.120] seconds\n","\n","0%   10   20   30   40   50   60   70   80   90   100%\n","|----|----|----|----|----|----|----|----|----|----|\n","***************************************************\n","\n","0%   10   20   30   40   50   60   70   80   90   100%\n","|----|----|----|----|----|----|----|----|----|----|\n","****************************************************Saving hnsw index @ : /content/gdrive/My Drive/FaceProcessing/code/output/vgg16_FacialFeatures.hnsw\n","Saving face to image mapping disctionary @: /content/gdrive/My Drive/FaceProcessing/code/output/vgg16_face_to_image.pkl\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"ikgC4Bvv3nmD","colab_type":"code","outputId":"7bd8f809-3351-44e5-941c-33d10a3fdc16","executionInfo":{"status":"ok","timestamp":1588366935734,"user_tz":-330,"elapsed":561994,"user":{"displayName":"Mahantesh ambi","photoUrl":"","userId":"14442223101884743872"}},"colab":{"base_uri":"https://localhost:8080/","height":1000}},"source":["!python /content/gdrive/My\\ Drive/FaceProcessing/code/FeatureExtractor.py -model_type \"resnet50\" -debug False -images /content/gdrive/My\\ Drive/FaceProcessing/processed_square_images/"],"execution_count":0,"outputs":[{"output_type":"stream","text":["Using TensorFlow backend.\n","2020-05-01 20:52:56.585979: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1\n","Your CPU supports instructions that this binary was not compiled to use: SSE3 SSE4.1 SSE4.2 AVX AVX2\n","For maximum performance, you can install NMSLIB from sources \n","pip install --no-binary :all: nmslib\n","2020-05-01 20:52:57.907537: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1\n","2020-05-01 20:52:57.933783: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-01 20:52:57.934378: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: \n","pciBusID: 0000:00:04.0 name: Tesla P4 computeCapability: 6.1\n","coreClock: 1.1135GHz coreCount: 20 deviceMemorySize: 7.43GiB deviceMemoryBandwidth: 178.99GiB/s\n","2020-05-01 20:52:57.934430: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1\n","2020-05-01 20:52:57.935873: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10\n","2020-05-01 20:52:57.937343: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10\n","2020-05-01 20:52:57.937673: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10\n","2020-05-01 20:52:57.939178: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10\n","2020-05-01 20:52:57.939886: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10\n","2020-05-01 20:52:57.943268: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7\n","2020-05-01 20:52:57.943378: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-01 20:52:57.944019: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-01 20:52:57.944630: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1703] Adding visible gpu devices: 0\n","2020-05-01 20:52:57.944877: I tensorflow/core/platform/cpu_feature_guard.cc:143] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX512F\n","2020-05-01 20:52:57.949185: I tensorflow/core/platform/profile_utils/cpu_utils.cc:102] CPU Frequency: 2000150000 Hz\n","2020-05-01 20:52:57.949380: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x181d640 initialized for platform Host (this does not guarantee that XLA will be used). Devices:\n","2020-05-01 20:52:57.949409: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version\n","2020-05-01 20:52:58.037752: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-01 20:52:58.038322: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x6818000 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:\n","2020-05-01 20:52:58.038359: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Tesla P4, Compute Capability 6.1\n","2020-05-01 20:52:58.038545: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-01 20:52:58.038908: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: \n","pciBusID: 0000:00:04.0 name: Tesla P4 computeCapability: 6.1\n","coreClock: 1.1135GHz coreCount: 20 deviceMemorySize: 7.43GiB deviceMemoryBandwidth: 178.99GiB/s\n","2020-05-01 20:52:58.038967: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1\n","2020-05-01 20:52:58.039001: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10\n","2020-05-01 20:52:58.039024: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcufft.so.10\n","2020-05-01 20:52:58.039045: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcurand.so.10\n","2020-05-01 20:52:58.039065: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusolver.so.10\n","2020-05-01 20:52:58.039100: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcusparse.so.10\n","2020-05-01 20:52:58.039121: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7\n","2020-05-01 20:52:58.039191: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-01 20:52:58.039560: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-01 20:52:58.039879: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1703] Adding visible gpu devices: 0\n","2020-05-01 20:52:58.039940: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudart.so.10.1\n","2020-05-01 20:52:58.557759: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1102] Device interconnect StreamExecutor with strength 1 edge matrix:\n","2020-05-01 20:52:58.557818: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1108]      0 \n","2020-05-01 20:52:58.557832: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1121] 0:   N \n","2020-05-01 20:52:58.558059: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-01 20:52:58.558517: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:981] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero\n","2020-05-01 20:52:58.558901: W tensorflow/core/common_runtime/gpu/gpu_bfc_allocator.cc:39] Overriding allow_growth setting because the TF_FORCE_GPU_ALLOW_GROWTH environment variable is set. Original config value was 0.\n","2020-05-01 20:52:58.558946: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1247] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 6966 MB memory) -> physical GPU (device: 0, name: Tesla P4, pci bus id: 0000:00:04.0, compute capability: 6.1)\n","2020-05-01 20:52:58.814769: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7\n","2020-05-01 20:52:59.874129: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10\n","(337, -27, 474, 124) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000014.jpg\n","(171, -13, 223, 50) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000035.jpg\n","(-9, 110, 66, 210) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000057.jpg\n","(79, -13, 401, 405) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000050.jpg\n","(383, 66, 513, 226) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000049.jpg\n","(-4, 113, 117, 280) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000049.jpg\n","(397, -29, 532, 151) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000074.jpg\n","(427, -8, 512, 111) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000072.jpg\n","(251, -27, 333, 68) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000072.jpg\n","(303, -20, 399, 88) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000075.jpg\n","Extracted 100 faces in [61.381] seconds \n","(390, 133, 520, 326) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000110.jpg\n","(369, 65, 522, 303) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000170.jpg\n","(-7, 83, 87, 228) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000177.jpg\n","(-21, 59, 151, 305) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000204.jpg\n","Extracted 200 faces in [63.367] seconds \n","(440, 31, 517, 130) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000215.jpg\n","(-7, 86, 122, 279) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000230.jpg\n","(-5, 53, 30, 112) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000250.jpg\n","(410, 58, 513, 198) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000242.jpg\n","(301, -13, 395, 102) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000247.jpg\n","(258, 371, 386, 530) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000245.jpg\n","(343, -1, 392, 60) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000264.jpg\n","(253, -6, 305, 60) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000260.jpg\n","(315, 62, 512, 340) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000300.jpg\n","Extracted 300 faces in [57.721] seconds \n","(-9, 112, 184, 370) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000320.jpg\n","(395, 435, 470, 525) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000380.jpg\n","Extracted 400 faces in [67.101] seconds \n","(-9, 89, 211, 408) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000419.jpg\n","(334, 90, 538, 398) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000419.jpg\n","(392, 462, 433, 517) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000430.jpg\n","(89, -7, 147, 60) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000423.jpg\n","(-1, -10, 36, 33) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000423.jpg\n","(-14, 91, 49, 178) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000423.jpg\n","(-12, 74, 140, 283) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000436.jpg\n","(-9, 232, 63, 359) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000467.jpg\n","(-1, 316, 32, 365) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000485.jpg\n","Extracted 500 faces in [66.818] seconds \n","(-5, 92, 229, 385) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000500.jpg\n","(291, 47, 519, 350) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000500.jpg\n","(-11, 106, 109, 289) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000493.jpg\n","(466, 170, 528, 267) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000512.jpg\n","(245, -5, 287, 40) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000530.jpg\n","(-5, 179, 96, 315) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000533.jpg\n","(108, -21, 189, 69) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000578.jpg\n","(205, -17, 506, 406) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000574.jpg\n","(457, 140, 513, 218) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000588.jpg\n","Extracted 600 faces in [65.401] seconds \n","(297, -14, 384, 96) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000600.jpg\n","(-5, 90, 94, 240) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000630.jpg\n","(-7, 131, 116, 311) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000659.jpg\n","(468, 82, 514, 149) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000678.jpg\n","MTCNN did not detect face for: /content/gdrive/My Drive/FaceProcessing/processed_square_images/000699.jpg\n","(219, 481, 258, 528) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000711.jpg\n","Extracted 700 faces in [64.196] seconds \n","(425, 91, 529, 249) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000755.jpg\n","(-2, 107, 79, 211) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000783.jpg\n","(475, 231, 524, 312) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000793.jpg\n","(173, -26, 440, 300) is beyond image of size: (512, 512, 3) for file /content/gdrive/My Drive/FaceProcessing/processed_square_images/000814.jpg\n","Extracted 800 faces in [65.021] seconds \n","MTCNN did not detect face for: /content/gdrive/My Drive/FaceProcessing/processed_square_images/000836.jpg\n","Extracted faces in 839 images at [535.586] seconds\n","Total 1530 faces detected\n","Preprocessed faces in [0.879] seconds: \n","Extracted features in [13.589] seconds\n","\n","0%   10   20   30   40   50   60   70   80   90   100%\n","|----|----|----|----|----|----|----|----|----|----|\n","***************************************************\n","\n","0%   10   20   30   40   50   60   70   80   90   100%\n","|----|----|----|----|----|----|----|----|----|----|\n","****************************************************Saving hnsw index @ : /content/gdrive/My Drive/FaceProcessing/code/output/l2_resnet50_FacialFeatures.hnsw\n","Saving face to image mapping disctionary @: /content/gdrive/My Drive/FaceProcessing/code/output/l2_resnet50_face_to_image.pkl\n"],"name":"stdout"}]}]}